dbms/src/Server/Server.cpp

```diff
@@ -242,9 +242,29 @@ struct TiFlashProxyConfig
             else
                 args_map["advertise-engine-addr"] = args_map["engine-addr"];
             args_map["engine-label"] = getProxyLabelByDisaggregatedMode(disaggregated_mode);
-            // For tiflash write node, it should report a extra label with "key" == "engine-role-label"
+            String extra_label;
             if (disaggregated_mode == DisaggregatedMode::Storage)
-                args_map["engine-role-label"] = DISAGGREGATED_MODE_WRITE_ENGINE_ROLE;
+            {
```

Comment:
Wrap it with `#if SERVERLESS_PROXY == 1`?

Reply:
@CalvinNeo PTAL again